README.md

```diff
@@ -4,222 +4,1008 @@
+
+Let's have a quick look at the help text first, to see the changes being reflected:
+
+```bash
+$> ./mytool --help
+Usage: DEFAULT {0/number} [mode]
+
+Description:
+    The default command.
+
+Positional Args:
+    0,number                     - The number to check.
+
+Named Args:
+    --mode                       - Which mode to use.
+```
+
+And now let's test our functionality:
+
+```bash
+# JCLI supports most common argument styles.
+
+# Even (Normal)
+$> mytool 60 --mode normal
+Program exited with status code 1
+
+# Even (Reversed)
+$> mytool 60 --mode=reversed
+Program exited with status code 0
+
+# Bad value for mode
+$> mytool 60 --mode lol
+std.conv.ConvException@\src\phobos\std\conv.d(2817): Mode does not have a member named 'reverse'
+
+# Can safely assume Odd behaves properly.
+
+# Now, we haven't marked --mode as optional, so...
+$> mytool 60
+ERROR: The following required named arguments were not provided: ["mode"]
+Program exited with status code -1
+```
+
+We can see that `--mode` is working as expected, however notice that in the last case, the user isn't allowed to leave out `--mode` since it's not marked as optional.
+
+## Optional Arguments
+
+Optional arguments, as the name implies, are optional. Only Named arguments can be optional (technically, Positional arguments can be optional in certain use cases, but JCLI doesn't support that... yet).
+
+Inside of D's standard library - Phobos - is the module [std.typecons](https://dlang.org/phobos/std_typecons.html) which contains a type called [Nullable](https://dlang.org/phobos/std_typecons.html#Nullable).
+
+JCLI has special support for this type, as it is used to mark an argument as optional. This type is publicly imported by JCLI, so you don't have to import anything extra.
+
+Anyway, all we want to do is make our `mode` argument a `Nullable`, so JCLI knows it's optional:
 
-You can add the [JIOC](https://code.dlang.org/packages/jioc) library into your project, and then create and pass a `ServiceProvider` into the constructor
-of `CommandLineInterface`. Afterwards, all command objects are created with dependency injection via JIOC's `Injector.construct`.
+```d
+@Command(null, "The default command.")
+struct DefaultCommand
+{
+    @CommandPositionalArg(0, "number", "The number to check.")
+    int number;
+
+    @CommandNamedArg("mode", "Which mode to use.")
+    Nullable!Mode mode;
 
-This is the *only* way `CommandLineInterface` is able to pass data into a command's constructor.
+    int onExecute()
+    {
+        if(this.mode.get(Mode.normal) == Mode.normal)
+            return number % 2 == 0 ? 1 : 0;
+        else
+            return number % 2;
+    }
+}
+```
 
-Please see [this](https://github.com/SealabJaster/jcli/tree/master/examples/05-dependency-injection) example for more information.
+The first change is that `Mode mode;` has now become `Nullable!Mode mode`, to make it, quite literally, a `Mode` that is `Nullable`.
```

Comment:
Suggestion (instead of few lines here) to remove some basic things:

There are two changes:
- Type of `mode` member becomes `Nullable!Mode`.
- `onExecute` now uses `mode.get(Mode.normal)` which returns `Mode.normal` is `--mode` option is not provided.

Reply:
It was mostly to point out something that might be easy to glance over. I've removed the first line as its redundant with your above suggestion, but I've kept the second line.